src/algorand/algorandEngine.ts

```diff
@@ -604,6 +626,81 @@ export class AlgorandEngine extends CurrencyEngine<
   getDisplayPublicSeed(): string {
     return this.walletInfo.keys.publicKey ?? ''
   }
+
+  engineGetActivationAssets = async (
+    options: EdgeEngineGetActivationAssetsOptions
+  ): Promise<EdgeGetActivationAssetsResults> => {
+    return {
+      assetOptions: [
+        {
+          paymentWalletId: this.walletId,
+          currencyPluginId: this.currencyInfo.pluginId
+        }
+      ]
+    }
+  }
+
+  engineActivateWallet = async ({
+    activateTokenIds,
+    paymentTokenId,
+    paymentWallet
+  }: EdgeEngineActivationOptions): Promise<EdgeActivationQuote> => {
+    if (activateTokenIds == null)
+      throw new Error(
+        `Must specify activateTokenIds for ${this.currencyInfo.currencyCode}`
+      )
+    if (paymentTokenId != null)
+      throw new Error(`Must activate with ${this.currencyInfo.currencyCode}`)
+    if (paymentWallet?.id !== this.walletId)
+      throw new Error('Must pay with same wallet you are activating token with')
+
+    const approvalTransactions: EdgeTransaction[] = []
+
+    for (const activateTokenId of activateTokenIds) {
+      if (this.allTokensMap[activateTokenId] == null)
+        throw new Error(`Invalid tokenId to activate ${activateTokenId}`)
+
+      const spendInfo: EdgeSpendInfo = {
+        currencyCode: this.allTokensMap[activateTokenId].currencyCode,
+        skipChecks: true,
+        spendTargets: [
+          { publicAddress: this.walletInfo.keys.publicKey, nativeAmount: '0' }
+        ] // Activation ("opt-in") transactions are just asset transfers with a zero amount
+      }
+      const approvalTx = await this.makeSpend(spendInfo)
```

Comment:
As mentioned in the GUI PR, I would like to move the `catch` logic here so we still return a quote, even if we have insufficient funds for the `makeSpend` call.